### PR TITLE
Fix #29

### LINKED ISSUE #29: Log fail status code
Is there a way to log the status code for a fail? So instead of getting `:status=>"fail"` I would like to see `:status=>405` or whatever.

### PR DESCRIPTION
**Pull request by Strnadj**
Not sure about **serialized_response_body**, it is designed to always return a **Hash**? In case of Internal server error it returns string with error message...

### CODE REVIEW

**Comment by aserafin** [comment]
thanks @Strnadj :) I will review this and hopefully merge during the weekend

**Comment by eMandistribution** [comment]
ping @aserafin  ;-)

**Comment by Strnadj** [comment]
ping @aserafin :-)

**Comment by nnluukhtn** [comment]
Hi @aserafin, do you have any plan to merge this PR?

**Comment by aserafin** [comment]
hey @nnluukhtn - yes, I do :) but this not trivial change and I have to come up with some time to review and test this. If you want to help and try this branch in your application and report back it would be awesome :)

**Comment by Strnadj** [comment]
We are using this on three projects and no problems ...  yet... :-)

**Comment by aserafin** [comment]
good to know @Strnadj :) did you merged latest changes from master or running just your branch?

**Comment by Strnadj** [comment]
Unfortunatelly Iam using my fork.

**Comment by bluegod** [comment]
It would be great to get this merged 👍

**Comment by Tolsto** [comment]
We also use this patch, works great so far. Please consider merging.

File: lib/grape_logging/middleware/request_logger.rb [hunk 1]
```diff
@@ -23,34 +27,71 @@ def initialize(app, options = {})
...
-        begin
-          super
-        rescue
+        # Catch error
+        error = catch(:error) do
+          begin
+            @app_response = @app.call(@env)
+          rescue => e
+            # Log as 500 + message
+            after(500, e.message)
```
Comment (maciej-szlosarczyk) [inline lib/grape_logging/middleware/request_logger.rb]: Hi @Strnadj, this is not correct. It always logs 500, even if the actual error code is different. For example, for `Grape::Exceptions::MethodNotAllowed` it logs:

```
{"status":500,"time":{"total":39.52,"db":3.81,"view":35.71},"method":"DELETE","path":"/papi/ping","params":{},"host":"localhost"}
```

It should be:
```
{"status":405,"time":{"total":39.52,"db":3.81,"view":35.71},"method":"DELETE","path":"/papi/ping","params":{},"host":"localhost"}
```

Change to `after(e.status, e.message)` fixes the issue.

Comment (Strnadj) [inline lib/grape_logging/middleware/request_logger.rb]: This make sence, but.... method `status` is not on all RuntimeExceptions catched by this part of code... What about `e.respond_to?(:status) ? e.status : 500`.

Comment (Strnadj) [inline lib/grape_logging/middleware/request_logger.rb]: Or something similar

Comment (maciej-szlosarczyk) [inline lib/grape_logging/middleware/request_logger.rb]: Yeah, that is indeed better.

**Pull request merged**